### PR TITLE
Feat/unusual investigations home

### PR DESCRIPTION
**Pull request by barnahum8**
- unusual investigations card. 
- in progress (with sub status in progress) over 4 hours investigations filter.
- completed with no contacts investigations filter.
- added card to home screen.
- statistics query edit.

User Story: [1286](https://dev.azure.com/spectrumFactory/CoronaI/_sprints/backlog/CoronaI%20Team/CoronaI/Sprint%2014?workitem=1286)

### CODE REVIEW

File: client/src/commons/statusToFilterConvertor.ts [hunk 1]
```diff
@@ -27,6 +27,17 @@ const statusToFilterConvertor = {
         inactiveUserFilter: true,
         unassignedUserFilter: false
     },
+    [FilterRulesDescription.UNUSUAL_IN_PROCESS]: {
+        statusFilter: [InvestigationMainStatusCodes.IN_PROCESS],
+        updateDateFilter: new Date(Date.now() - (4 * 60 * 60 * 1000)).toUTCString(),
```
Comment (nevobonneh) [inline client/src/commons/statusToFilterConvertor.ts:32]: maybe use a const instead

File: client/src/components/App/Content/LandingPage/adminLandingPage/PostponedCard/PostponedCardStyles.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import { makeStyles } from '@material-ui/styles';
 
+export const cardWidth = '14vw';
 export const cardHeight = '16vh';
 
 const useStyles = makeStyles({
     postponedCard: {
+        width: cardWidth,
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/adminLandingPage/PostponedCard/PostponedCardStyles.ts:8]: I think you should drop it and set width in the grid containing it instead (md)

File: client/src/components/App/Content/LandingPage/adminLandingPage/UnallocatedCard/UnallocatedCardStyles.ts [hunk 3]
```diff
@@ -1,9 +1,11 @@
 import { makeStyles } from '@material-ui/styles';
 
 export const cardHeight = '8vh';
+export const cardWidth = '14vw';
 
 const useStyles = makeStyles({
     unallocatedCard: {
+        width: cardWidth,
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/adminLandingPage/UnallocatedCard/UnallocatedCardStyles.ts:8]: same here

File: client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    const { onCompleteClick, onInProcessClick, isLoading, unusualInProgressInvestigationsCount, unusualCompletedNoContactInvestigationsCount} = props;
+
+    return (
+        <LoadingCard isLoading={isLoading} width={cardWidth} height={cardHeight} className={classes.unusualCard}>
+            <div className={classes.unusualTitle}>
+                <WarningIcon className={classes.warningIcon}></WarningIcon>
+                <Typography className={classes.unusualTitleText}><b>לתשומת ליבך</b></Typography>
+            </div>
+            <Tooltip className={[classes.unusualCompleted, hoverClasses.whiteButtons].join(' ')} title={unusualCompletesInvestigationsText}>
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx:29]: I think you should think over the use of hoverClasses.whiteButtons at this location, cursor: 'pointer' is being used and not all parts in your card are clickable

File: client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                        <Typography className={classes.investigationAmountText}><b>חקירות</b></Typography>
+                    </div>
+                    <div onClick={() => onCompleteClick(statusToFilterConvertor[FilterRulesDescription.UNUSUAL_COMPLETED_NO_CONTACT])} className={classes.unusualInvestigations}>
+                        <Typography className={classes.unusualInvestigationsText}><b>{FilterRulesDescription.UNUSUAL_COMPLETED_NO_CONTACT}</b></Typography>
+                        <NavigateBeforeIcon className={classes.navigateIcon} />
+                    </div>
+                </div>
+            </Tooltip>
+            <Divider></Divider>
+            <Tooltip className={[classes.unusualInProcess, hoverClasses.whiteButtons].join(' ')} title={unusualInProcessInvestigationsText}>
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx:42]: same here

File: client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                        <Typography className={classes.investigationNumberText}><b>{unusualCompletedNoContactInvestigationsCount}</b></Typography>
+                        <Typography className={classes.investigationAmountText}><b>חקירות</b></Typography>
+                    </div>
+                    <div onClick={() => onCompleteClick(statusToFilterConvertor[FilterRulesDescription.UNUSUAL_COMPLETED_NO_CONTACT])} className={classes.unusualInvestigations}>
+                        <Typography className={classes.unusualInvestigationsText}><b>{FilterRulesDescription.UNUSUAL_COMPLETED_NO_CONTACT}</b></Typography>
+                        <NavigateBeforeIcon className={classes.navigateIcon} />
+                    </div>
+                </div>
+            </Tooltip>
+            <Divider></Divider>
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx:41]: use <Divider/ > instead

File: client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+            <Tooltip className={[classes.unusualInProcess, hoverClasses.whiteButtons].join(' ')} title={unusualInProcessInvestigationsText}>
+                <div>
+                    <div className={classes.investigationAmount}>
+                        <Typography className={classes.investigationNumberText}><b>{unusualInProgressInvestigationsCount}</b></Typography>
+                        <Typography className={classes.investigationAmountText}><b>חקירות</b></Typography>
+                    </div>
+                    <div onClick={() => onInProcessClick(statusToFilterConvertor[FilterRulesDescription.UNUSUAL_IN_PROCESS])} className={classes.unusualInvestigations}>
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx:48]: I think you should change **onIn**ProcessClick name

File: client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCardStyles.ts [hunk 8]
```diff
@@ -0,0 +1,58 @@
+import { makeStyles } from '@material-ui/styles';
+
+export const cardWidth = '14vw';
+export const cardHeight = '38vh';
+
+const useStyles = makeStyles({
+    unusualCard: {
+        width: cardWidth,
+        height: cardHeight,
```
Comment (nevobonneh) [inline client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCardStyles.ts:9]: I think dropping both width and height from here will be better (defining both using xs and md in the grid will help you control the component size better)

**Review by nevobonneh — CHANGES_REQUESTED** [review]

File: client/src/commons/statusToFilterConvertor.ts (hunk 1 above)
Comment (barnahum8) [reply client/src/commons/statusToFilterConvertor.ts]: 👍

File: client/src/components/App/Content/LandingPage/adminLandingPage/PostponedCard/PostponedCardStyles.ts (hunk 2 above)
Comment (barnahum8) [reply client/src/components/App/Content/LandingPage/adminLandingPage/PostponedCard/PostponedCardStyles.ts]: 👍

File: client/src/components/App/Content/LandingPage/adminLandingPage/UnallocatedCard/UnallocatedCardStyles.ts (hunk 3 above)
Comment (barnahum8) [reply client/src/components/App/Content/LandingPage/adminLandingPage/UnallocatedCard/UnallocatedCardStyles.ts]: 👍

File: client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx (hunk 4 above)
Comment (barnahum8) [reply client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx:29]: removed cursor instead.

File: client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx (hunk 5 above)
Comment (barnahum8) [reply client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx:42]: 👍

File: client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx (hunk 6 above)
Comment (barnahum8) [reply client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx]: 👍

File: client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx (hunk 7 above)
Comment (barnahum8) [reply client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx]: 👍

**Review by nevobonneh — APPROVED** [review]

**Review by nevobonneh — APPROVED** [review]

File: client/src/components/App/Content/LandingPage/InvestigationTable/TableFilter/TableFilter.tsx [hunk 9]
```diff
@@ -23,7 +23,7 @@ const TableFilter = (props: Props) => {
         onFilterChange, onClose, 
         changeInactiveUserFilter, onSubStatusChange, inactiveUserFilter, 
         changeUnassginedUserFilter, unassignedUserFilter, 
-        timeRangeFilter, onTimeRangeFilterChange
+        timeRangeFilter, onTimeRangeFilterChange, updateDateFilter, nonContactFilter
```
Comment (Noam6421) [inline client/src/components/App/Content/LandingPage/InvestigationTable/TableFilter/TableFilter.tsx:26]: maybe break the row (put  updateDateFilter, nonContactFilter in new row)

File: client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx [hunk 10]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { Tooltip, Typography, Divider } from '@material-ui/core';
+import WarningIcon from '@material-ui/icons/Warning';
```
Comment (Noam6421) [inline client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx:3]: replace rows 2 and 3

File: client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+export default UnusualCard;
+
+interface Props {
+    onUnusualCompletedNoContactInvestigationsClick: (infoFilter: FilterRulesVariables) => void;
+    onUnusualInProgressInvestigationsClick: (infoFilter: FilterRulesVariables) => void;
+    isLoading: boolean;
+    unusualInProgressInvestigationsCount: number;
+    unusualCompletedNoContactInvestigationsCount: number;
+
```
Comment (Noam6421) [inline client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx:66]: remove empty row

File: client/src/components/App/Content/LandingPage/adminLandingPage/adminLandingPage.tsx [hunk 12]
```diff
@@ -10,6 +10,7 @@ import useStyles from './adminLandingPageStyles';
 import useAdminLandingPage from './useAdminLandingPage';
 import PostponedCard from './PostponedCard/PostponedCard';
 import UnallocatedCard from './UnallocatedCard/UnallocatedCard';
+import UnusualCard from './UnusualCard/UnusualCard';
```
Comment (Noam6421) [inline client/src/components/App/Content/LandingPage/adminLandingPage/adminLandingPage.tsx:13]: move up the import to line 10

**Review by Noam6421 — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/LandingPage/InvestigationTable/TableFilter/TableFilter.tsx (hunk 9 above)
Comment (barnahum8) [reply client/src/components/App/Content/LandingPage/InvestigationTable/TableFilter/TableFilter.tsx]: 👍

File: client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx (hunk 10 above)
Comment (barnahum8) [reply client/src/components/App/Content/LandingPage/adminLandingPage/UnusualCard/UnusualCard.tsx:2]: 👍

File: client/src/components/App/Content/LandingPage/adminLandingPage/adminLandingPage.tsx (hunk 12 above)
Comment (barnahum8) [reply client/src/components/App/Content/LandingPage/adminLandingPage/adminLandingPage.tsx]: 👍

**Review by Noam6421 — APPROVED** [review]

**Review by nevobonneh — APPROVED** [review]

**Pull request merged**